File: leetcode/3541. Find Most Frequent Vowel and Consonant/main.ts
const maxFreqSum = (s: string): number => {
  let [maxVowel, maxConsonant] = [0, 0];

  let vowels = new Set<string>(['a', 'e', 'i', 'o', 'u'])
  let frequency = new Array(26).fill(0);

  for (let char of s) {
    let idx = char.charCodeAt(0) - 'a'.charCodeAt(0);
    if (idx < 0 || idx > 26) continue;

    frequency[idx] += 1;

    if (vowels.has(char)) {
      maxVowel = Math.max(maxVowel, frequency[idx])
    } else {
      maxConsonant = Math.max(maxConsonant, frequency[idx])
    }
  }

  return maxVowel + maxConsonant;
};

File: leetcode/3541. Find Most Frequent Vowel and Consonant/main.rs
impl Solution {
    pub fn max_freq_sum(s: String) -> i32 {
        let mut vowels = [0; 26];
        let mut consonants = [0; 26];

        let (mut max_vowel, mut max_consonant) = (0, 0);

        for ch in s.chars() {
            if ch < 'a' || ch > 'z' {
                continue;
            }

            let idx = (ch as u8 - b'a') as usize;

            match ch {
                'a' | 'e' | 'i' | 'o' | 'u' => {
                    vowels[idx] += 1;
                    max_vowel = max_vowel.max(vowels[idx]);
                }
                _ => {
                    consonants[idx] += 1;
                    max_consonant = max_consonant.max(consonants[idx]);
                }
            }
        }

        max_vowel + max_consonant
    }
}
